lua/diffview/config.lua

```diff
@@ -182,6 +182,11 @@ M.defaults = {
       { "n", "[x",            actions.prev_conflict,        { desc = "Go to the previous conflict" } },
       { "n", "]x",            actions.next_conflict,        { desc = "Go to the next conflict" } },
       { "n", "g?",            actions.help("file_panel"),   { desc = "Open the help panel" } },
+      { "n", "<leader>co",    actions.conflict_choose_file("ours"),    { desc = "Choose the OURS version of a conflict" } },
+      { "n", "<leader>ct",    actions.conflict_choose_file("theirs"),  { desc = "Choose the THEIRS version of a conflict" } },
+      { "n", "<leader>cb",    actions.conflict_choose_file("base"),    { desc = "Choose the BASE version of a conflict" } },
+      { "n", "<leader>ca",    actions.conflict_choose_file("all"),     { desc = "Choose all the versions of a conflict" } },
+      { "n", "dx",            actions.conflict_choose_file("none"),    { desc = "Delete the conflict region" } },
```

Comment:
These should not be file panel mappings, because the action doesn't operate on the entry under the cursor. It operates on the currently open file in the view. So these mappings should be moved to the `view` context.

I also think the actual mappings should be:

```
<leader>cO
<leader>cT
<leader>cB
<leader>cA
dX
```

Reply:
I changed to the mappings you provided, they make more sense.

Having the bindings in the `view` means I can't trigger it from the `file_panel`, which I also would like to be able to do. For me it makes sense to have them in both, same as for `]x` which is also in both.

I added them to both now and can remove if you disagree.

Reply:
Wanting to be able to use the action from the file panel is fine, but you haven't implemented the action as a file panel action. Notably it does not behave like other file panel actions that operate on file entries. Specifically: it doesn't operate on the entry under the cursor in the file panel, but rather *always* on the entry currently open in the view.

As a user I would experience it as highly unexpected behavior if I put the cursor on file `foo` in the file panel, invoked an action, and then had the action operate on `bar` instead, simply because that was the file open in the view.

Our rules for file inference are also clearly defined under `:h diffview-file-inference`.

---

If you want to make it a file panel action it needs to change in a few ways:

- If the panel is *not* focused: the action does exactly what it currently does.
- If the panel *is* focused:
    - Get the current item under the cursor in the panel
    - If it's the currently open file: the action does exactly what it currently does.
    - If it's *not* the currently open file:
        - The file entry needs to be opened
        - We then await the event that signifies that the file is ready (opening file entries is an async operation because we might need to create the files from Git).
        - Then the action proceeds like normal.

To save you the hassle of figuring out the correct API's to call here, I'll give you the file opening logic:

```lua
local view = lib.get_current_view()
if not (view and view:instanceof(DiffView.__get())) then return end
---@cast view DiffView

if view.panel:is_focused() then
  local item = view:infer_cur_file(false) ---@cast item -DirData
  if not item then return end

  if item.active then
    -- The entry is already open and the action can proceed like normal
  else
    -- The entry is not open

    view.emitter:once("file_open_post", function(e, ...)
      -- When this callback is invoked, the entry is ready and the action can
      -- proceed like normal
    end)

    -- Open the entry
    view:set_file(item)
  end
end
```

Reply:
You are right, when I think about it I would also expect it to operate on the file under the cursor then on the active view, so I will read the help page you provided and change the implementation to fit the actual use case.

Many thanks for taking time to write me the file opening logic part, really appreciate it!